cpg_workflows/metamist.py

```diff
@@ -760,3 +764,8 @@ def parse_reads(  # pylint: disable=too-many-return-statements
             )
 
         return fastq_pairs
+
+
+def filter_analyses_by_meta(analyses: list[dict], meta_filter: dict) -> list[dict]:
+    """Filter analyses by matching key-value pairs in the meta field."""
+    return [a for a in analyses if a.get("meta") and all(a["meta"].get(k) == v for k, v in meta_filter.items())]
```

Comment:
How did the double quotes get past the linter-whiner…

Reply:
looks like we have it turned off https://github.com/populationgenomics/production-pipelines/blob/main/pyproject.toml#L3C1-L3C26

Reply:
whooops

Reply:
This is a pretty frustrating one.. AFAIK this is because we're skipping this rule with Black, and also skipping this rule ([Q000](https://github.com/populationgenomics/production-pipelines/blob/main/pyproject.toml#L66)) with Ruff. The `quote-style = "single"` rule is set in ruff.format, but we don't run ruff-format in CI, only ruff-lint.

Basically, I'm not sure this repo has had its grand reckoning PR when incorporating ruff (instead of/in addition to black), so there's still a lot of broken linting rules. Best to just look past some of these and solve when code is migrated out